.github/workflows/doc-tests.yml

```diff
@@ -9,22 +9,37 @@ on:
 permissions:
   contents: read
 
+env:
+  HATCH_ENV_TYPE_VIRTUAL_PATH: ${{ github.workspace }}/.hatch_envs
+
 jobs:
   doc-test:
-    name: Sphinx-${{ matrix.builder }}
+    name: Sphinx Documentation Tests
     runs-on: ubuntu-latest
-    continue-on-error: ${{ matrix.experimental || false }}
-    strategy:
-      fail-fast: false
-      matrix:
-        builder: [ html, linkcheck ]
-        include:
-          # Run default html builder with warnings as error
-          - builder: html
-            args: -W
+
     steps:
       - uses: actions/checkout@v4
-      - name: Run sphinx builder ${{ matrix.builder }}
-        run: |
-          pip install hatch
-          hatch run docs:${{ matrix.builder }} ${{ matrix.args }}
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+          cache: 'pip'
+
+      - name: Cache Hatch environments
+        uses: actions/cache@v4
+        with:
+          path: |
+            ${{ github.workspace }}/.hatch_envs
+          key: hatch-envs-${{ hashFiles('pyproject.toml') }}
+          restore-keys: |
+            hatch-envs-
```

Comment:
Can you test if you update an optional-dependencies, and you run `hatch run ...`, does hatch update the environment?

Reply:
hmm, good call: https://github.com/teemtee/tmt/actions/runs/11233762115/job/31228137758 
This was supposed to be 5min patch lol

Reply:
Interesting way to check :sweat_smile:. Seems that was not the case, so just remove this extra `restore-keys`.